numpy/core/function_base.py

```diff
@@ -91,38 +92,31 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None):
     if dtype is None:
         dtype = dt
 
-    if num <= 0:
-        return array([], dtype)
-    if num == 1:
-        return array([start], dtype=dtype)
     y = _nx.arange(0, num, dtype=dt)
-    if endpoint:
-        num -= 1
-    y /= num
-    y *= stop - start
+
+    if num > 1:
+        delta = stop - start
+        step = delta / div
+        if step == 0:
+            # Special handling for denormal numbers, gh-5437
+            y /= div
+            y *= delta
+        else:
+            y *= step
+    else:
+        # 0 and 1 item long sequences have an undefined step
```

Comment:
It could, but it doesn't right now. A negative `num` is also a good candidate for a `ValueError`, but it also never has.

There was some minimal discussion on the list between Chris Barker and Ben Root, on whether to return whatever or raise an error. I have no opinion either way, but in the absence of a strong argument, I would go with the principle of least surprise and leave things as they are.